catalog/resourcemanagement/resourcemanagement-querymonitor-ui/src/main/java/org/codice/ddf/resourcemanagement/query/service/QueryMonitor.java

```diff
@@ -0,0 +1,115 @@
+import org.codice.ddf.resourcemanagement.query.plugin.QueryMonitorPlugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryMonitor implements QueryMonitorMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryMonitor.class);
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    private QueryMonitorPlugin queryMonitorPlugin;
+
+    public QueryMonitor(QueryMonitorPlugin queryMonitorPlugin) {
+        this.queryMonitorPlugin = queryMonitorPlugin;
+        registerMbean();
+    }
+
+    private void registerMbean() {
+        try {
+            objectName = new ObjectName(QueryMonitor.class.getName() + ":service=querymonitor");
+            mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        } catch (MalformedObjectNameException e) {
+            LOGGER.error("Unable to create Query Monitor Configuration MBean.", e);
+        }
+        if (mBeanServer == null) {
+            return;
+        }
+        try {
+            try {
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Registered Query Monitor Configuration MBean under object name: {}",
+                        objectName.toString());
+            } catch (InstanceAlreadyExistsException e) {
+                // Try to remove and re-register
+                mBeanServer.unregisterMBean(objectName);
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Re-registered Query Monitor Configuration MBean");
+            }
+        } catch (MBeanRegistrationException | InstanceNotFoundException |
+                InstanceAlreadyExistsException | NotCompliantMBeanException e) {
+            LOGGER.error("Could not register MBean [{}].", objectName.toString(), e);
+        }
+    }
+
+    @Override
+    public List<Map<String, String>> activeSearches() {
+        List<Map<String, String>> activeSearchList = new ArrayList<>();
+        Map<UUID, ActiveSearch> activeSearchMap = queryMonitorPlugin.getActiveSearches();
+
+        if (MapUtils.isNotEmpty(activeSearchMap)) {
+            for (Map.Entry<UUID, ActiveSearch> entry : activeSearchMap.entrySet()) {
+                ActiveSearch activeSearch = entry.getValue();
+                Map<String, String> map = new HashMap<>();
+                map.put("user", activeSearch.getClientInfo());
+                map.put("sourceId",
+                        activeSearch.getSource()
+                                .getId());
+                map.put("query", activeSearch.getCQL());
+
+                SimpleDateFormat dt = new SimpleDateFormat("dd MMM yyyy hh:mm:ss");
+                Date date = activeSearch.getStartTime();
+                String dateString = dt.format(date);
+                map.put("dateStarted", dateString);
+                map.put("uuid",
+                        activeSearch.getUniqueID()
+                                .toString());
```

Comment:
Should all of the string literals used in the map.put() calls  be defined as constants?